app/mixins/downloadable.js

```diff
@@ -1,19 +1,36 @@
 import Mixin from '@ember/object/mixin';
-import { inject as service } from '@ember/service';
 import nestProfile from '../utils/nest-profile';
 
+/**
+ * The Downloadable mixin is an EmberJS feature, see https://api.emberjs.com/ember/release/classes/Mixin.
+ *
+ * Downloadable defines a collection of common functionality used across the profile/* routes. It is injected
+ * into every one of those routes. In this case, it does two unrelated things:
+ *  - Update the profile view's tab state, so that the tab is visually highlighted
+ *  - It "nests" the incoming profile data and replaces original array structure
+ *    with the nested structure on the controller, necessary for profile data display, downloading
+ *    and charts.
+ */
 export default Mixin.create({
-  selection: service(),
-
+  /**
+   * EmberJS Route beforeModel hook.
+   * Here this is used to unload the store to avoid the issue with duplicate ids.
+   * TODO: There are duplicates because id is based arbitrarily on the array index.
```

Comment:
does this mean that if we had different IDs, we could cache all the profiles for a given session and not have to redownload when selecting back & forth btwn multiple profiles? we should add this to a todo ticket :)

Reply:
https://github.com/NYCPlanning/labs-factfinder/issues/711

@julialucyhogan